e2e/cp_test.go

```diff
@@ -2926,3 +2926,407 @@ func TestCopyS3ToDirDryRun(t *testing.T) {
+
+	result.Assert(t, icmd.Success)
+
+	assertLines(t, result.Stdout(), map[int]compareFunc{
+		0: equals(`cp %vreadme.md %vreadme.md`, src, dst),
+	})
+
+	// assert local filesystem
+	expected := fs.Expected(t, folderLayout...)
+	assert.Assert(t, fs.Equal(workdir.Path(), expected))
+
+	expectedS3Content := map[string]string{
+		"prefix/readme.md": "this is a readme file",
+	}
+
+	nonExpectedS3Content := map[string]string{
+		"prefix/b/filename-with-hypen.gz": "file has hypen in its name",
+		"prefix/a/another_test_file.txt":  "yet another txt file. yatf.",
+		"prefix/testfile1.txt":            "this is a test file 1",
+	}
+
+	// assert objects should be in S3
+	for key, content := range expectedS3Content {
+		assert.Assert(t, ensureS3Object(s3client, bucket, key, content))
+	}
+
+	//assert objects should not be in S3.
+	for key, content := range nonExpectedS3Content {
+		err := ensureS3Object(s3client, bucket, key, content)
+		assertError(t, err, errS3NoSuchKey)
+	}
+}
+
+// cp --exclude "main*" 's3://srcbucket/*' s3://dstbucket
+func TestCopySingleS3ObjectsIntoAnotherBucketWithExcludeFilter(t *testing.T) {
+	t.Parallel()
+
+	const (
+		srcbucket = "bucket"
+		dstbucket = "dstbucket"
+	)
+
+	s3client, s5cmd, cleanup := setup(t)
+	defer cleanup()
+
+	createBucket(t, s3client, srcbucket)
+	createBucket(t, s3client, dstbucket)
+
+	files := []string{
+		"file.txt",
+		"file1.txt",
+		"main.py",
+		"main.js",
+		"readme.md",
+		"main.pdf",
+		"main/file.txt",
+	}
+
+	expectedFiles := []string{
+		"file.txt",
+		"file1.txt",
+		"readme.md",
+	}
+
+	nonExpectedFiles := []string{
+		"main.py",
+		"main.js",
+		"main.pdf",
+		"main/file.txt",
+	}
+
+	const (
+		content        = "this is a file content"
+		excludePattern = "main*"
+	)
+
+	for _, filename := range files {
+		putFile(t, s3client, srcbucket, filename, content)
+	}
+
+	src := fmt.Sprintf("s3://%v/*", srcbucket)
+	dst := fmt.Sprintf("s3://%v/", dstbucket)
+
+	cmd := s5cmd("cp", "--exclude", excludePattern, src, dst)
+	result := icmd.RunCmd(cmd)
+
+	result.Assert(t, icmd.Success)
+
+	assertLines(t, result.Stdout(), map[int]compareFunc{
+		0: equals(`cp s3://%s/file.txt s3://%s/file.txt`, srcbucket, dstbucket),
+		1: equals(`cp s3://%s/file1.txt s3://%s/file1.txt`, srcbucket, dstbucket),
+		2: equals(`cp s3://%s/readme.md s3://%s/readme.md`, srcbucket, dstbucket),
+	}, sortInput(true))
+
+	// assert s3 source objects
+	for _, filename := range files {
+		assert.Assert(t, ensureS3Object(s3client, srcbucket, filename, content))
+	}
+
+	// assert s3 destination objects
+	for _, filename := range expectedFiles {
+		assert.Assert(t, ensureS3Object(s3client, dstbucket, filename, content))
+	}
+
+	// assert s3 destination objects which should not be in bucket.
+	for _, filename := range nonExpectedFiles {
+		err := ensureS3Object(s3client, dstbucket, filename, content)
+		assertError(t, err, errS3NoSuchKey)
+	}
+}
+
+func TestCopySingleS3ObjectsIntoAnotherBucketWithExcludeFilters(t *testing.T) {
```

Comment:
Could you add a sample command as doc like the above tests?

Reply:
added new usage case to `cp`.